config/config_test.go

```diff
@@ -764,9 +682,203 @@ func TestMigrateConfigFromEnv(t *testing.T) {
+	}
+
+	if oldval, ok := os.LookupEnv("PBS_ADAPTERS_BIDDER1_APP_SECRET"); ok {
+		defer os.Setenv("PBS_ADAPTERS_BIDDER1_APP_SECRET", oldval)
+	} else {
+		defer os.Unsetenv("PBS_ADAPTERS_BIDDER1_APP_SECRET")
+	}
+
+	if oldval, ok := os.LookupEnv("PBS_ADAPTERS_BIDDER1_ENDPOINTCOMPRESSION"); ok {
+		defer os.Setenv("PBS_ADAPTERS_BIDDER1_ENDPOINTCOMPRESSION", oldval)
+	} else {
+		defer os.Unsetenv("PBS_ADAPTERS_BIDDER1_ENDPOINTCOMPRESSION")
+	}
+
+	if oldval, ok := os.LookupEnv("PBS_ADAPTERS_BIDDER1_USERSYNC_REDIRECT_URL"); ok {
+		defer os.Setenv("PBS_ADAPTERS_BIDDER1_USERSYNC_REDIRECT_URL", oldval)
+	} else {
+		defer os.Unsetenv("PBS_ADAPTERS_BIDDER1_USERSYNC_REDIRECT_URL")
+	}
+
+	// set new
+	os.Setenv("PBS_ADAPTERS_BIDDER1_DISABLED", "true")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_ENDPOINT", "http://some.url/override")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_EXTRA_INFO", `{"extrainfo": true}`)
+
+	os.Setenv("PBS_ADAPTERS_BIDDER1_MAINTAINER_EMAIL", "maintainer@some.url")
+
+	os.Setenv("PBS_ADAPTERS_BIDDER1_DEBUG_ALLOW", "true")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_GVLVENDORID", "42")
+
+	os.Setenv("PBS_ADAPTERS_BIDDER1_EXPERIMENT_ADSCERT_ENABLED", "true")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_CAPABILITIES_APP_MEDIATYPES", "video,native")
+
+	os.Setenv("PBS_ADAPTERS_BIDDER1_XAPI_USERNAME", "username_override")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_XAPI_PASSWORD", "password_override")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_XAPI_TRACKER", "tracker_override")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_PLATFORM_ID", "platform_id_override")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_APP_SECRET", "app_secret_override")
+	os.Setenv("PBS_ADAPTERS_BIDDER1_ENDPOINTCOMPRESSION", "GZIP")
+
+	os.Setenv("PBS_ADAPTERS_BIDDER1_USERSYNC_REDIRECT_URL", "http://some.url/sync?redirect={{.RedirectURL}}")
+
+	cfg, _ := newDefaultConfig(t)
+
+	assert.Equal(t, true, cfg.BidderInfos["bidder1"].Disabled)
+	assert.Equal(t, "http://some.url/override", cfg.BidderInfos["bidder1"].Endpoint)
+	assert.Equal(t, `{"extrainfo": true}`, cfg.BidderInfos["bidder1"].ExtraAdapterInfo)
+
+	assert.Equal(t, "maintainer@some.url", cfg.BidderInfos["bidder1"].Maintainer.Email)
+
+	assert.Equal(t, true, cfg.BidderInfos["bidder1"].Debug.Allow)
+	assert.Equal(t, uint16(42), cfg.BidderInfos["bidder1"].GVLVendorID)
+
+	assert.Equal(t, []openrtb_ext.BidType{openrtb_ext.BidTypeVideo, openrtb_ext.BidTypeNative}, cfg.BidderInfos["bidder1"].Capabilities.App.MediaTypes)
+
+	assert.Equal(t, true, cfg.BidderInfos["bidder1"].Experiment.AdsCert.Enabled)
+	assert.Equal(t, "username_override", cfg.BidderInfos["bidder1"].XAPI.Username)
+	assert.Equal(t, "password_override", cfg.BidderInfos["bidder1"].XAPI.Password)
+	assert.Equal(t, "tracker_override", cfg.BidderInfos["bidder1"].XAPI.Tracker)
+
+	assert.Equal(t, "platform_id_override", cfg.BidderInfos["bidder1"].PlatformID)
+	assert.Equal(t, "app_secret_override", cfg.BidderInfos["bidder1"].AppSecret)
+	assert.Equal(t, "GZIP", cfg.BidderInfos["bidder1"].EndpointCompression)
 }
```

Comment:
IMHO we don't need to test all of the env variables, just a few is fine to provide viper binding works ok. This seems a little too verbose IMHO.

Reply:
Removed some of checks